templates/blob-storage-to-dynamics-365-import.json

```diff
@@ -0,0 +1,379 @@
+											"source": "@body('Copy_file_to_processing_folder')?['Path']"
+										}
+									}
+								}
+							}
+						}
+					},
+					"default": {
+						"actions": {
+							"Copy_file_to_failed_folder": {
+								"runAfter": {},
+								"type": "ApiConnection",
+								"inputs": {
+									"host": {
+										"connection": {
+											"name": "@parameters('$connections')['azureblob']['connectionId']"
+										}
+									},
+									"method": "post",
+									"path": "/datasets/default/copyFile",
+									"queries": {
+										"destination": "/failed/@{body('Copy_file_to_processing_folder')?['Name']}",
+										"overwrite": true,
+										"queryParametersSingleEncoded": true,
+										"source": "@body('Copy_file_to_processing_folder')?['Path']"
+									}
+								}
+							}
+						}
+					},
+					"expression": "@body('Get_Execution_Results')['value']",
+					"type": "Switch"
+				},
+				"Send_email_if_API_returns_an_error": {
+					"runAfter": {
+						"Initiate_Package_Import": [
+							"Failed"
+						]
+					},
+					"type": "ApiConnection",
+					"inputs": {
+						"body": {
+							"Body": "D365 Import has failed using the API in Logic App BlobStorageToD365.\n\nFilename: @{body('Copy_file_to_processing_folder')?['Name']}",
+							"Subject": "Logic App Failure",
+							"To": ""
+						},
+						"host": {
+							"connection": {
+								"name": "@parameters('$connections')['office365-2']['connectionId']"
+							}
+						},
+						"method": "post",
+						"path": "/Mail"
+					}
+				},
+				"Wait_For_Execution_Results": {
+					"actions": {
+						"Delay": {
+							"runAfter": {
+								"Get_Execution_Results": [
+									"Succeeded"
+								]
+							},
+							"type": "Wait",
+							"inputs": {
+								"interval": {
+									"count": 15,
+									"unit": "Second"
+								}
+							}
+						},
+						"Get_Execution_Results": {
+							"runAfter": {},
+							"type": "Http",
+							"inputs": {
+								"authentication": {
+									"audience": "[YOUR_D365_INSTANCE_HERE].dynamics.com",
+									"clientId": "",
+									"secret": "",
+									"tenant": "microsoft.com",
+									"type": "ActiveDirectoryOAuth",
+									"authority": ""
+								},
+								"body": {
+									"executionId": "@body('Initiate_Package_Import')['value']"
+								},
+								"method": "POST",
+								"uri": "[YOUR_D365_INSTANCE_HERE].dynamics.com/data/DataManagementDefinitionGroups/Microsoft.Dynamics.DataEntities.GetExecutionSummaryStatus"
+							}
+						}
+					},
+					"runAfter": {
+						"Initiate_Package_Import": [
+							"Succeeded"
+						]
+					},
+					"expression": "@or(equals(body('Get_Execution_Results')?['value'], 'Succeeded'), equals(body('Get_Execution_Results')?['value'], 'PartiallySucceeded'), equals(body('Get_Execution_Results')?['value'], 'Canceled'), equals(body('Get_Execution_Results')?['value'], 'Failed'))",
+					"limit": {
+						"count": 60,
```

Comment:
Since you're waiting 15 seconds with the delay, the count limit will almost always be hit first (15 minutes vs. 1h), make sure that's the behavior you want.